File: smk_scm/public/py/material_request.py
import frappe
from frappe.email.queue import flush
from frappe.utils import get_url

@frappe.whitelist()
def send_email(name, doctype, company, recipient_id, recipient, mr_details):
    logo_url = get_url("/files/SMK logo.jpg")
    doctype_slug = frappe.scrub(doctype).replace("_", "-")
    document_url = frappe.utils.get_url(f"app/{doctype_slug}/{name}")
    message = f"""
    <p>Dear {recipient},</p>
    <p>I hope this message finds you well.
    <br>We are currently in need of the following items for our operations and would like to request a quotation for the same.
    <br>Click <a href="{document_url}">here</a> to open the Material Request.</p>
    <b>Item Details:</b>
    <p>{mr_details}</p>
    <p>Please feel free to contact us if any further details are needed.
    <br>Thank you for your prompt attention to this request. We look forward to your response.</p>
    <p>Best regards,<br>{company}</p>
    <img src="{logo_url}" alt="SMK Petrochemicals" width="200" />
    """
    frappe.sendmail(
        recipients=[recipient_id],
        subject="Material Request - '" + name + "'",
        message=message
    )
    flush()

File: smk_scm/public/js/material_request.js
frappe.ui.form.on("Material Request", {
    on_submit(frm) {
        let mr_details = `
            <table border="1" cellpadding="5" cellspacing="0">
                <thead>
                    <tr>
                        <th>Item Name</th>
                        <th>Quantity</th>
                        <th>UOM</th>
                        <th>Required By</th>
                        <th>Description</th>
                    </tr>
                </thead>
                <tbody>
        `;
        frm.doc.items.forEach(item => {
            let descriptionText = item.description.replace(/<\/?p[^>]*>/g, '').trim();
            var inputDate = item.schedule_date;
            var parts = inputDate.split("-");
            var year = parts[0];
            var month = parts[1];
            var day = parts[2];
            var formattedDate = day ? `${day}-${month}-${year}` : `${year}`;
            mr_details += `
                <tr>
                    <td>${item.item_name}</td>
                    <td>${item.qty}</td>
                    <td>${item.uom}</td>
                    <td>${formattedDate}</td>
                    <td>${descriptionText}</td>
                </tr>
            `;
        });
        mr_details += `
                </tbody>
            </table>
        `;
        frappe.call({
            method: 'smk_scm.public.py.material_request.send_email',
            args: {
                name: frm.doc.name,
                doctype: frm.doc.doctype,
                company: frm.doc.company,
                recipient_id: frm.doc.custom_user_id || "",
                recipient: frm.doc.custom_user_full_name || "",
                mr_details
            },
            callback: function(response) {
                if (response.message) {
                    frappe.msgprint('An Email sent successfully');
                }
            }
        });
    }
});
